### PR TITLE
Restrict pyspark version for pip install; doc update

### PR DESCRIPTION
**Pull request by hkvision**
Fix https://github.com/intel-analytics/zoo/issues/336
As this example will fail for spark 2.3.1.

### CODE REVIEW

File: pyzoo/setup.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def setup_package():
         license='Apache License, Version 2.0',
         url='https://github.com/intel-analytics/analytics-zoo',
         packages=packages,
-        install_requires=['bigdl==0.6.0'],
+        install_requires=['pyspark<2.3.0', 'bigdl==0.6.0'],
```
Comment (jason-dai) [inline pyzoo/setup.py]: shall we specify pyspark >2.2.0?

Comment (hkvision) [reply pyzoo/setup.py]: updated.

**Comment by hkvision** [comment]
Synced with @jason-dai offline, for this release, we restrict pyspark<2.3. In the future, after we test against spark 2.3, we can remove this constraint.

**Comment by hkvision** [comment]
Issued created for spark 2.3 https://github.com/intel-analytics/zoo/issues/338

**Pull request merged**